backend/tests/legacy/test_api_allocations.py

```diff
@@ -0,0 +1,154 @@
+def test_allocations(
+    client: Client, deployer: UserAccount, ua_alice: UserAccount, ua_bob: UserAccount
+):
+    alice_proposals = get_proposals_addresses(1)[:3]
+
+    # fund Octant
+    deployer.fund_octant(
+        address=client.config["WITHDRAWALS_TARGET_CONTRACT_ADDRESS"], value=400
+    )
+
+    # lock GLM from two accounts
+    deployer.transfer(ua_alice, 10000)
+    ua_alice.lock(10000)
+    deployer.transfer(ua_bob, 15000)
+    ua_bob.lock(15000)
+
+    # forward time to the beginning of the epoch 2
+    client.move_to_next_epoch(2)
+
+    # wait for indexer to catch up
+    epoch_no = client.wait_for_sync(2)
+    print(f"indexed epoch: {epoch_no}")
+
+    # make a snapshot
+    res = client.pending_snapshot()
+    assert res["epoch"] > 0
+
+    ua_alice.allocate(1000, alice_proposals)
+    ua_bob.allocate(1000, alice_proposals[:1])
+
+    allocations, _ = client.get_epoch_allocations(1)
+    unique_donors = set()
+    unique_proposals = set()
+    print("Allocations: \n", allocations)
+
+    assert len(allocations["allocations"]) == 4
+    for index, allocation in enumerate(allocations["allocations"]):
+        for key, val in allocation.items():
+            if key == "donor":
+                unique_donors.add(val)
+            if key == "amount":
+                assert int(val) > 0
+            if key == "proposal":
+                unique_proposals.add(val)
+        print(
+            "Allocation ",
+            index + 1,
+            ": ",
+            allocation,
+        )
+
+    assert len(unique_donors) == 2
+    assert len(unique_proposals) == 3
+
+
+@pytest.mark.api
+def test_allocations_basics(
+    client: Client, deployer: UserAccount, ua_alice: UserAccount, ua_bob: UserAccount
+):
+    alice_proposals = get_proposals_addresses(1)[:3]
+
+    # fund Octant
+    deployer.fund_octant(
+        address=client.config["WITHDRAWALS_TARGET_CONTRACT_ADDRESS"], value=400
+    )
+
+    # lock GLM from one account
+    deployer.transfer(ua_alice, 10000)
+    ua_alice.lock(10000)
+
+    # forward time to the beginning of the epoch 2
+    client.move_to_next_epoch(2)
+
+    # wait for indexer to catch up
+    epoch_no = client.wait_for_sync(2)
+    print(f"indexed epoch: {epoch_no}")
+
+    # make a snapshot
+    res = client.pending_snapshot()
+    assert res["epoch"] > 0
+
+    # Making allocations
+    nonce, status_code = client.get_allocation_nonce(ua_alice.address)
+    # Nonce is always 0
+    assert status_code == 200, "Nonce status code is different than 200"
+
+    allocation_amount = 1000
+    allocation_response_code = ua_alice.allocate(allocation_amount, alice_proposals)
+    assert (
+        allocation_response_code == 201
+    ), "Allocation status code is different than 201"
+
+    epoch_allocations, status_code = client.get_epoch_allocations(1)
+    assert len(epoch_allocations["allocations"]) == len(alice_proposals)
+
+    for index, allocation in enumerate(epoch_allocations["allocations"]):
```

Comment:
```suggestion
    for allocation in epoch_allocations["allocations"]:
    assert allocation["donor"] == ua_alice.address, "Donor address is wrong"
    assert int(allocation["amount"]) == allocation_amount
    assert allocation["proposal"], "Proposal address is empty"
    print("Allocation: ", allocation)
```

Reply:
Valid comment, thank you. Is there anything against enumerate for logging purposes?